services/auth/source/oauth2/providers_simple.go

```diff
@@ -31,7 +31,10 @@ type SimpleProvider struct {
 
 // CreateGothProvider creates a GothProvider from this Provider
 func (c *SimpleProvider) CreateGothProvider(providerName, callbackURL string, source *Source) (goth.Provider, error) {
-	return c.newFn(source.ClientID, source.ClientSecret, callbackURL, c.scopes...), nil
+	scopes := make([]string, len(c.scopes)+len(source.Scopes))
+	copy(scopes, c.scopes)
+	copy(scopes[len(c.scopes):], source.Scopes)
```

Comment:
This notation is forced by our linter... previously this was `scopes := append(c.scopes, source.Scopes...)` but the linter doesn't like that because it assigns the result of append to a different slice. 

:man_shrugging:

Reply:
linter is correct.

```
func main() {
	a := make([]int, 0, 10)
	a = append(a, 1)
	b := append(a, 2)
	log.Printf("a=%v, b=%v", a, b)
	log.Printf("a=%p, b=%p", a, b)
	b[0] = 100
	log.Printf("a=%v, b=%v", a, b)
}
```

Otherwise `a` and `b` share the same memory and it may lead to bugs.